frontend/src/search/SearchResult.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import { Link, useSearchParams } from "react-router-dom";
+import { getUsers } from "../user/user-service";
+import User from "../user/user";
+import { useGlobalContext } from "../auth/GlobalContext";
+import { format } from "date-fns";
+
+const SearchResult: React.FC = () => {
+  const [searchParams] = useSearchParams();
+  const { refresh } = useGlobalContext();
+  const [results, setResults] = useState<User[]>([]);
+
+  useEffect(() => {
+    const resultHandler = async () => {
+      const query = searchParams.get("q");
+      if (!query) return;
+
+      const res = await getUsers();
```

Comment:
Instead of getting all users and filtering, write a findUsers function instead in the backend.